ipyleaflet/velocity.py

```diff
@@ -38,49 +38,68 @@ class Velocity(Layer):
-    _model_name = Unicode('LeafletVelocityModel').tag(sync=True)
+    _view_name = Unicode("LeafletVelocityView").tag(sync=True)
+    _model_name = Unicode("LeafletVelocityModel").tag(sync=True)
 
-    zonal_speed = Unicode('', help='Name of the zonal speed in the dataset')
-    meridional_speed = Unicode('', help='Name of the meridional speed in the dataset')
-    latitude_dimension = Unicode('latitude', help='Name of the latitude dimension in the dataset')
-    longitude_dimension = Unicode('longitude', help='Name of the longitude dimension in the dataset')
+    zonal_speed = Unicode("", help="Name of the zonal speed in the dataset")
+    meridional_speed = Unicode("", help="Name of the meridional speed in the dataset")
+    latitude_dimension = Unicode(
+        "latitude", help="Name of the latitude dimension in the dataset"
+    )
+    longitude_dimension = Unicode(
+        "longitude", help="Name of the longitude dimension in the dataset"
+    )
     units = Unicode(None, allow_none=True)
 
     data = Dataset().tag(dtype=None, sync=True, to_json=ds_x_to_json)
 
     # Options
     display_values = Bool(True).tag(sync=True, o=True)
-    display_options = Dict({
-        'velocityType': 'Global Wind',
-        'position': 'bottomleft',
-        'emptyString': 'No velocity data',
-        'angleConvention': 'bearingCW',
-        'displayPosition': 'bottomleft',
-        'displayEmptyString': 'No velocity data',
-        'speedUnit': 'kt'
-    }).tag(sync=True)
+    display_options = Dict(
+        {
+            "velocityType": "Global Wind",
+            "position": "bottomleft",
+            "emptyString": "No velocity data",
+            "angleConvention": "bearingCW",
+            "displayPosition": "bottomleft",
+            "displayEmptyString": "No velocity data",
+            "speedUnit": "kt",
+        }
+    ).tag(sync=True)
     min_velocity = Float(0).tag(sync=True, o=True)
     max_velocity = Float(10).tag(sync=True, o=True)
     velocity_scale = Float(0.005).tag(sync=True, o=True)
-    color_scale = List([
-        "rgb(36,104, 180)",
-        "rgb(60,157, 194)",
-        "rgb(128,205,193)",
-        "rgb(151,218,168)",
-        "rgb(198,231,181)",
-        "rgb(238,247,217)",
-        "rgb(255,238,159)",
-        "rgb(252,217,125)",
-        "rgb(255,182,100)",
-        "rgb(252,150,75)",
-        "rgb(250,112,52)",
-        "rgb(245,64,32)",
-        "rgb(237,45,28)",
-        "rgb(220,24,32)",
-        "rgb(180,0,35)"
-    ]).tag(sync=True, o=True)
+    colormap = Any(linear.OrRd_06)
+    color_scale = List([]).tag(sync=True, o=True)
+    caption = Unicode("").tag(sync=True, o=True)
+
+    @default("color_scale")
+    def _default_color_scale(self):
+        self.color_scale = []
+
+        for color in self.colormap.colors:
+            rgb_tuple = tuple(int(x * 256) for x in color[:3])
+            rgb_str = f"rgb{rgb_tuple}"
+            self.color_scale.append(rgb_str)
+
+        return self.color_scale
```

Comment:
Some suggestions/issues here:
- you should not set `self.color_scale` yourself (same comment as for `subitems`)
- we must multiply by `255` because we scale the range from `[0., 1.]` to `[0, 255]` (256 colors)
- we can use list comprehension here instead of a for-loop
- we can/should probably use `rgba` if that's how the colormap is defined

```suggestion
        return [
            f"rgba{tuple(int(x * 255) for x in color)}"
            for color in self.colormap.colors
        ]
```